File: TodoList.Application/Services/TodoItem/ITodoService.cs
using TodoList.Core.DTOs.TodoItem;

namespace TodoList.Core.Services.TodoItem;

public interface ITodoService
{
    Task<TodoItemDto> CreateTodoAsync(CreateTodoItemDto dto, int userId);
    Task DeleteTodoAsync(int todoId, int userId);
    Task<TodoListDto> GetUserTodoListAsync(int userId);
    Task<TodoItemDto> ToggleTodoCompletionAsync(int todoId, int userId);
}

File: TodoList.Client/src/app/todo/services/ITodoService.ts
import {CreateTodoItemDto} from '../models/CreateTodoItemDto';
import {Observable} from 'rxjs';
import {TodoItemDto} from '../models/TodoItemDto';
import {TodoListDto} from '../models/TodoListDto';

export interface ITodoService {
  createTodo(dto: CreateTodoItemDto): Observable<TodoItemDto>;
  deleteTodo(todoId: number): Observable<void>;
  getUserTodoList(): Observable<TodoListDto>;
  toggleTodoCompletion(todoId: number): Observable<TodoItemDto>;
}
